src/shadow-embed.js

```diff
@@ -58,16 +59,27 @@ export function createShadowRoot(hostElement) {
     return existingRoot;
   }
 
+  // Define our shadowRoot object to be returned
+  let shadowRoot;
+
   // Native support.
   const shadowDomSupported = getShadowDomSupportedVersion();
   if (shadowDomSupported == ShadowDomVersion.V1) {
-    return hostElement.attachShadow({mode: 'open'});
+    shadowRoot = hostElement.attachShadow({mode: 'open'});
   } else if (shadowDomSupported == ShadowDomVersion.V0) {
-    return hostElement.createShadowRoot();
+    shadowRoot = hostElement.createShadowRoot();
+  } else {
+    // Polyfill.
+    shadowRoot = createShadowRootPolyfill(hostElement);
   }
 
-  // Polyfill.
-  return createShadowRootPolyfill(hostElement);
+  // Ensure that Shadow CSS is supported, and add an id if not
+  if (!isShadowCssSupported()) {
+    const randomId = Math.floor(Math.random() * 10000);
+    shadowRoot.id = `i-amphtml-sd-${randomId}`;
```

Comment:
Done.

I kept the `shadowRoot.id = rootId;` assignment, as we still need a way of passing the id around on the element. And I figured it would be better than iterating through the class list, to find the id. Let me know if that makes sense.